plugins/panels/similarity_search/operators.py

```
@@ -236,6 +247,76 @@ def _compute_combined_query(
+
+        # Server-side size limit aligned with UI constraint (10 MB raw)
+        max_bytes = 10 * 1024 * 1024
+        if len(content) > ((max_bytes * 4) // 3) + 8:
+            raise ValueError(
+                "Uploaded image '%s' exceeds size limit for brain key '%s'"
+                % (name, brain_key)
+            )
+
+        try:
+            img_bytes = base64.b64decode(content)
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)